src/psbt.h

```diff
@@ -27,26 +28,30 @@ static constexpr uint8_t PSBT_IN_WITNESSSCRIPT = 0x05;
 static constexpr uint8_t PSBT_IN_BIP32_DERIVATION = 0x06;
 static constexpr uint8_t PSBT_IN_SCRIPTSIG = 0x07;
 static constexpr uint8_t PSBT_IN_SCRIPTWITNESS = 0x08;
+static constexpr uint8_t PSBT_IN_PROPRIETARY = 0xFC;
 // Confidential Assets stuff (private use area)
-static constexpr uint8_t PSBT_IN_VALUE = 0xf0;
-static constexpr uint8_t PSBT_IN_VALUE_BLINDER = 0xf1;
-static constexpr uint8_t PSBT_IN_ASSET = 0xf2;
-static constexpr uint8_t PSBT_IN_ASSET_BLINDER = 0xf3;
+static constexpr uint8_t PSBT_IN_VALUE = 0x00;
+static constexpr uint8_t PSBT_IN_VALUE_BLINDER = 0x01;
+static constexpr uint8_t PSBT_IN_ASSET = 0x02;
+static constexpr uint8_t PSBT_IN_ASSET_BLINDER = 0x03;
 
 // Output types
 static constexpr uint8_t PSBT_OUT_REDEEMSCRIPT = 0x00;
 static constexpr uint8_t PSBT_OUT_WITNESSSCRIPT = 0x01;
 static constexpr uint8_t PSBT_OUT_BIP32_DERIVATION = 0x02;
+static constexpr uint8_t PSBT_OUT_PROPRIETARY = 0xFC;
 // Confidential Assets stuff (private use area)
-static constexpr uint8_t PSBT_OUT_VALUE_COMMITMENT = 0xf0;
-static constexpr uint8_t PSBT_OUT_VALUE_BLINDER = 0xf1;
-static constexpr uint8_t PSBT_OUT_ASSET_COMMITMENT = 0xf2;
-static constexpr uint8_t PSBT_OUT_ASSET_BLINDER = 0xf3;
-static constexpr uint8_t PSBT_OUT_RANGE_PROOF = 0xf4;
-static constexpr uint8_t PSBT_OUT_SURJECTION_PROOF = 0xf5;
-static constexpr uint8_t PSBT_OUT_BLINDING_PUBKEY = 0xf6;
-static constexpr uint8_t PSBT_OUT_NONCE_COMMITMENT = 0xf7;
-
+static constexpr uint8_t PSBT_OUT_VALUE_COMMITMENT = 0x00;
+static constexpr uint8_t PSBT_OUT_VALUE_BLINDER = 0x01;
+static constexpr uint8_t PSBT_OUT_ASSET_COMMITMENT = 0x02;
+static constexpr uint8_t PSBT_OUT_ASSET_BLINDER = 0x03;
+static constexpr uint8_t PSBT_OUT_RANGE_PROOF = 0x04;
+static constexpr uint8_t PSBT_OUT_SURJECTION_PROOF = 0x05;
+static constexpr uint8_t PSBT_OUT_BLINDING_PUBKEY = 0x06;
+static constexpr uint8_t PSBT_OUT_NONCE_COMMITMENT = 0x07;
+
+// Proprietary type identifer string
+static const std::string proprietary_id("elements");
```

Comment:
non-blocking nit: `PSBT_ELEMENTS_ID` or something

Reply:
I would also have made it a constant.

Reply:
Changed. Not quite sure what you mean by "make it a constant", it's already const.

Reply:
"formatted as a constant" sorry :)